### PR TITLE
feat(diff): add a baselineBranch option

### LINKED ISSUE #247: Configuration of the default branch to run diffs against
As a user, I can configure per repository the default branch the tests will be done against.

### Motivation

Material-UI use `next` as the default branch.

### PR DESCRIPTION
**Pull request by oliviertassinari**
Closes #247

### CODE REVIEW

File: src/server/models/Repository.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default class Repository extends BaseModel {
       'name',
       'enabled',
       'private',
+      'baselineBranch',
```
Comment (neoziro) [inline src/server/models/Repository.js]: not required if default

File: src/modules/synchronizer/GitHubSynchronizer.js [hunk 2]
```diff
@@ -44,6 +44,7 @@ class GitHubSynchronizer {
       } else {
         repository = await Repository.query().insert({
           ...data,
+          baselineBranch: 'master',
```
Comment (neoziro) [inline src/modules/synchronizer/GitHubSynchronizer.js]: Only one default please

File: src/server/test/factory.js [hunk 3]
```diff
@@ -62,6 +62,7 @@ factory.define('Repository', Repository, {
   githubId: factory.sequence('repository.githubId', n => n),
   name: factory.sequence('repository.name', n => `repo-${n}`),
   enabled: true,
+  baselineBranch: 'master',
```
Comment (neoziro) [inline src/server/test/factory.js]: Only one default

**Comment by neoziro** [comment]
I understand your point of view, but for me it's dangerous because if you forgot to change the database one the day you change the default, your application is no longer coherent. In fact it increases the change of having an incoherent state in the database.

**Comment by neoziro** [comment]
@oliviertassinari for this example is not very true, but in general I think it is.

**Comment by oliviertassinari** [comment]
I'm gonna remove the default databse

**Comment by neoziro** [comment]
Good for me.

**Pull request merged**